cdap-ui/app/features/hydrator/services/detail/stores/pipeline-detail-nodeconfig-store.js

```diff
@@ -37,7 +37,17 @@ angular.module(PKG.name + '.feature.hydrator')
     this.setState = function(plugin) {
       onPluginChange.call(this, plugin);
     };
+    this.setPlugin = function(plugin) {
+      this.plugin = plugin;
+      var type = this.isSource ? 'source': false;
```

Comment:
this code seems weird, since it seems like there's a boolean for isSource as well as isSink.
why not have just a 'type' variable on the `this` object? That seems simpler than the two booleans?